staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/generic/policy_dispatcher.go

```diff
@@ -194,23 +188,67 @@ func (d *policyDispatcher[P, B, E]) Dispatch(ctx context.Context, a admission.At
 					Evaluator: hook.Evaluator,
 				})
 			}
+		}
+	}
 
-			// VersionedAttr result will be cached and reused later during parallel
-			// hook calls
-			_, err = versionedAttrAccessor.VersionedAttribute(matchGVK)
-			if err != nil {
-				return apierrors.NewInternalError(err)
-			}
+	if len(relevantHooks) > 0 {
+		extraPolicyErrors, statusError := d.delegate(ctx, a, o, versionedAttrAccessor, relevantHooks)
+		if statusError != nil {
+			return statusError
+		}
+		policyErrors = append(policyErrors, extraPolicyErrors...)
+	}
+
+	var filteredErrors []PolicyError
+	for _, e := range policyErrors {
+		// we always default the FailurePolicy if it is unset and validate it in API level
+		var policy v1.FailurePolicyType
+		if fp := e.Policy.GetFailurePolicy(); fp == nil {
+			policy = v1.Fail
+		} else {
+			policy = *fp
 		}
 
+		switch policy {
+		case v1.Ignore:
+			// TODO: add metrics for ignored error here
+			continue
+		case v1.Fail:
+			filteredErrors = append(filteredErrors, e)
+		default:
+			filteredErrors = append(filteredErrors, e)
+		}
 	}
 
-	if len(relevantHooks) == 0 {
-		// no matching hooks
-		return nil
+	if len(filteredErrors) > 0 {
+		var err *apierrors.StatusError
+		if !errors.As(admission.NewForbidden(a, fmt.Errorf("admission request denied by policy")), &err) {
+			return apierrors.NewInternalError(fmt.Errorf("failed to create status error"))
+		}
+		err.ErrStatus.Message = ""
```

Comment:
What is this logic about? Reading `NewForbidden` it either returns `err` if it is already a forbidden error (it isn't here), or it returns `apierrors.NewInternalError(...)` or `apierrors.NewForbidden(...)`. In either case it is a status error. If we don't want to change `admission.NewForbidden`'s signature (maybe the best direction, but it has ripples), then:
```suggestion
		err, ok := admission.NewForbidden(...)
		if !ok {
			return apierrors.NewInternalError(fmt.Errorf("failed to create status error"))
		}
		err.ErrStatus.Message = ""
```
A lot easier to read.

Reply:
The linter didn't like:

```go
err, ok := admission.NewForbidden(a, fmt.Errorf("admission request denied by policy")).(*apierrors.StatusError)
if !ok {
	return apierrors.NewInternalError(fmt.Errorf("failed to create status error"))
}
```

```
ERROR: staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/generic/policy_dispatcher.go:229:14: type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors (errorlint)
ERROR: 		err, ok := admission.NewForbidden(a, fmt.Errorf("admission request denied by policy")).(*apierrors.StatusError)
ERROR: 		           ^

Please review the above warnings. You can test via "./hack/verify-golangci-lint.sh -r 43ad87b85a39344db31323b59c2eab4adc73b66e -n"
```

So I've restructured this a bit and added comments to make it clearer.